src/Microsoft.OData.Edm/VersioningList.cs

```diff
@@ -68,7 +68,7 @@ public override int Count
 
             public override VersioningList<TElement> Add(TElement value)
             {
-                return new LinkedVersioningList(this, value);
+                return new ArrayVersioningList(this, value);

```

Comment:
If I understand the implementation, this results in a new array being allocated, and elements copied from the previous array to the new one. Doesn't this lead to a larger memory footprint and more GC over time? Is the rationale that we're willing to sacrifice memory in order to get faster search times?